File: 645-set-mismatch/set-mismatch.js
var findErrorNums = function(nums) {
    let len = nums.length;
    let totalSum = (len * (len + 1)) / 2;
    let totalSumSq = (len * (len + 1) * (2 * len + 1)) / 6;
    let sum = 0, sumSq = 0;

    for(let num of nums) {
        sum += num;
        sumSq += num * num;
    }

    let diff = sum - totalSum;
    let sumDiff = (sumSq - totalSumSq) / diff;

    let duplicate = (diff + sumDiff) / 2;
    let missing = (sumDiff - diff) / 2;

    return [duplicate, missing];
};

File: 645-set-mismatch/set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int len = nums.size();
        long totalSum = (long)len * (len + 1) / 2;
        long totalSumSq = (long)len * (len + 1) * (2 * len + 1) / 6;
        long sum = 0, sumSq = 0;

        for(int num : nums) {
            sum += num;
            sumSq += (long)num * num;
        }

        long diff = sum - totalSum;
        long sumDiff = (sumSq - totalSumSq) / diff;

        int duplicate = (diff + sumDiff) / 2;
        int missing = (sumDiff - diff) / 2;

        return {duplicate, missing};
    }
};
